mkosi/config.py

```diff
@@ -3358,19 +3358,45 @@ def resolve_deps(images: Sequence[argparse.Namespace], include: Sequence[str]) -
     return sorted(images, key=lambda i: order.index(i.image))
 
 
-def parse_config(argv: Sequence[str] = (), *, resources: Path = Path("/")) -> tuple[Args, tuple[Config, ...]]:
+class ConfigAction(argparse.Action):
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[Any], None],
+        option_string: Optional[str] = None
+    ) -> None:
+        assert option_string is not None
+
+        if values is None and self.nargs == "?":
+            values = self.const or "yes"
 
-    class ParseContext:
+        try:
+            s = SETTINGS_LOOKUP_BY_DEST[self.dest]
+        except KeyError:
+            die(f"Unknown setting {option_string}")
```

Comment:
Shouldn't this be an assert?

Reply:
I just dropped the exception handling since we should never hit it in the first place